<?xml version="1.0" ?>
<launch>
    <machine name="apc-arm" address="apc-arm" env-loader="/home/apc/ros_ws/load_environment.sh"
        default="false" user="apc"/>

    <arg name="camera_name" default="realsense_wrist" />

    <node machine="apc-arm" name="acrv_$(arg camera_name)_ros" pkg="acrv_realsense_ros" type="acrv_realsense_ros_node" output="screen" respawn="true">
        <param name="serial_number" type="string" value="617205003472" />
        <param name="rotate_image_180" type="bool" value="false" />
        <param name="camera_name" type="string" value="$(arg camera_name)" />
        <!-- <param name="rgb_camera_info_path" type="string" value="/home/apc/ros_ws/src/acrv_apc_2017/acrv_cameras/acrv_realsense_ros/cfg/$(arg camera_name)/rgb_camera_info.yaml" /> -->
        <param name="rgb_hd_camera_info_path" type="string" value="/home/apc/ros_ws/src/acrv_apc_2017/acrv_cameras/acrv_realsense_ros/cfg/$(arg camera_name)/rgb_hd_camera_info.yaml" />
        <param name="depth_camera_info_path" type="string" value="/home/apc/ros_ws/src/acrv_apc_2017/acrv_cameras/acrv_realsense_ros/cfg/$(arg camera_name)/depth_camera_info.yaml" />
    </node>

    <!-- <node pkg="tf" type="static_transform_publisher" name="cartesian_realsense_link_broadcaster_image_pipeline"
            args="0 -0.02426723 0 -1.57 -0.01579435 -1.57 /realsense_endpoint /$(arg camera_name)_depth_optical_frame 10" /> -->

    <node pkg="tf" type="static_transform_publisher" name="cartesian_realsense_link_broadcaster_image_pipeline"
            args="0 0 0 -1.57 0 -1.57 /realsense_endpoint /$(arg camera_name)_rgb_optical_frame 10" />

    <!-- <node pkg="tf" type="static_transform_publisher" name="$(arg camera_name)_optical_frames"
        args="-0.02426723 -0.00047571 -0.00023367 -0.0001231  -0.01946618 -0.01579435 /$(arg camera_name)_depth_optical_frame /$(arg camera_name)_rgb_optical_frame 10" /> -->

    <node pkg="tf" type="static_transform_publisher" name="$(arg camera_name)_optical_frames"
        args="0.02426723 0.00047571 0.00023367 0.0001231  0.01946618 0.01579435 /$(arg camera_name)_rgb_optical_frame /$(arg camera_name)_depth_optical_frame 100" />

</launch>
